plutus-use-cases/src/Plutus/Contracts/Tutorial/Escrow.hs

```diff
@@ -302,7 +302,11 @@ refund ::
 refund inst _escrow = do
     pk <- ownFirstPaymentPubKeyHash
     unspentOutputs <- utxosAt (Scripts.validatorAddress inst)
-    let flt _ ciTxOut = either id Ledger.datumHash (Tx._ciTxOutScriptDatum ciTxOut) == Ledger.datumHash (Datum (PlutusTx.toBuiltinData pk))
+    let flt _ ocTxOut = case ocTxOut of
+                          Tx.ScriptOffChainTxOut _vh _v dh _m_va _m_da _m_rs ->
+                            dh == Ledger.datumHash (Datum (PlutusTx.toBuiltinData pk))
+                          Tx.PublicKeyOffChainTxOut {} ->
+                            False
```

Comment:
Same here, it's not clear whether it's relevent that the txOut is protected by a script or pubkey.

Reply:
Here it's the easiest to check, `flt` is applied to the result of `unspentOutputs <- utxosAt (Scripts.validatorAddress inst)`.

Reply:
ok so I can change this to `\ocTxOut -> ocTxOutDatum ocTxOut == Just (Ledger.datumHash (Datum (PlutusTx.toBuiltinData pk)))` ?

Reply:
Yes